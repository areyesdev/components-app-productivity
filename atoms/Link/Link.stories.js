import { Link, styles, options } from '.'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../helpers/storybook'

const Template = getTemplate(Link, styles)
const ListTemplate = getListTemplate(Link, styles)

export default {
  title: 'Atoms/Link',
  component: Link,
  args: {
    children: 'Mouths Muil',
  },
  argTypes: {
    color: getOptionsArgTypes(options.colors),
    size: getOptionsArgTypes(options.sizes),
  },
}

export const Default = Template.bind({})

export const Colors = ListTemplate.bind({})
Colors.args = { items: options.colors.map((color) => ({ color })) }

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }
