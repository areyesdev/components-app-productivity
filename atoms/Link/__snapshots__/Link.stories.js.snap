// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Link Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <a
    className="link color-primary"
  >
    <p
      className="paragraph color-primary size-md weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
  <a
    className="link color-tertiary"
  >
    <p
      className="paragraph color-tertiary size-md weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
</div>
`;

exports[`Storyshots Atoms/Link Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <a
    className="link color-primary"
  >
    <p
      className="paragraph color-primary size-md weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
</div>
`;

exports[`Storyshots Atoms/Link Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <a
    className="link color-primary"
  >
    <p
      className="paragraph color-primary size-xs weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
  <a
    className="link color-primary"
  >
    <p
      className="paragraph color-primary size-sm weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
  <a
    className="link color-primary"
  >
    <p
      className="paragraph color-primary size-md weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
  <a
    className="link color-primary"
  >
    <p
      className="paragraph color-primary size-lg weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
</div>
`;
